internal/provider/privateinternetaccess/tokenutils.go

```diff
@@ -0,0 +1,72 @@
+package privateinternetaccess
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+func fetchToken(ctx context.Context, client *http.Client,
+	tokenType string, authFilePath string) (token string, err error) {
+	username, password, err := getOpenvpnCredentials(authFilePath)
+	if err != nil {
+		return "", fmt.Errorf("getting username and password: %w", err)
+	}
+
+	errSubstitutions := map[string]string{
+		url.QueryEscape(username): "<username>",
+		url.QueryEscape(password): "<password>",
+	}
+
+	var path string
+
+	switch tokenType {
+	case "client":
+		path = "/api/client/v2/token"
+	case "gtoken":
+		path = "/gtoken/generateToken"
+	default:
+		return "", fmt.Errorf("token type %q is not supported", tokenType)
```

Comment:
you could even panic here, since that would be a programming error:

```suggestion
		panic(fmt.Sprintf("token type %q is not supported", tokenType))
```